### PR TITLE
Homepage videolisting

### PR DESCRIPTION
**Pull request by vishalg8454**
This PR adds the VideoCard listing in the homepage. The Videos are fetched via an axios API call from the db stored locally. A chip bar on top enables filtering of videos based on the category.

The components added are-

- `ChipBar` at src/components/chipbar/ChipBar
- `VideoCard` at src/components/video-card/Videocard
 
The context added are-

- `video-context.js` at src/context/video-context.js

[live preview](https://deploy-preview-2--videostock.netlify.app/)

image

image

Would love your feedback and suggestions.

### CODE REVIEW

**Review by chanchal-panpaliya — APPROVED** [review]
When I hover on the card, the video card is not displayed.
image

File: src/components/video-card/VideoCard.module.css [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  min-width: 5rem;
+  max-width: 20rem;
+  width: 100%;
+  gap: 4px;
+  padding: 8px;
+  box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px,
+    rgba(0, 0, 0, 0.06) 0px 1px 2px 0px;
+}
+.videoCard:hover {
+  opacity: 80%;
```
Comment (chanchal-panpaliya) [inline src/components/video-card/VideoCard.module.css]: change -> opacity: 0.8;

Comment (vishalg8454) [reply src/components/video-card/VideoCard.module.css]: Ah, yes. Will correct this. I don't know why my Chrome did not show this behaviour.

File: src/components/chipbar/ChipBar.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  );
+};
+const ChipBar = ({ categories = [], activeCategory, onClickHandler }) => {
+  return (
+    <div className={styles.chipContainer}>
+      {categories.map((it) => (
+        <Chip
+          categoryName={it}
+          activeCategory={activeCategory}
+          key={it}
```
Comment (ranitasahaa1230) [inline src/components/chipbar/ChipBar.jsx:24]: Change `it` and Give a proper naming

File: src/context/video-context.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const VideoProvider = ({ children }) => {
+  const [videoList, setVideoList] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(`api/videos`);
+        setVideoList(response.data.videos);
+      } catch (error) {}
```
Comment (ranitasahaa1230) [inline src/context/video-context.js:16]: Display the error in the UI itself

**Review by ranitasahaa1230 — APPROVED** [review]
Code looks clean
Good folder structure
Added some suggestions, do look into it
Rest, looks good

File: src/context/video-context.js (hunk 3 above)
Comment (vishalg8454) [reply src/context/video-context.js:16]: will dot this later with a toast component.

**Pull request merged**